pkg/client/client_test.go

```diff
@@ -741,68 +741,49 @@ func testAttachSocketConnection(socketPath string) error {
 	}
 	defer conn.Close()
 
-	outputStream := io.WriteCloser(os.Stdout)
-	errorStream := io.WriteCloser(os.Stderr)
-
-	receiveStdout := make(chan error)
+	// This second connection should be cleaned-up automatically
 	go func() {
-		receiveStdout <- redirectResponseToOutputStreams(outputStream, errorStream, conn)
-		close(receiveStdout)
+		conn, err := client.DialLongSocket("unixpacket", socketPath)
+		if err != nil {
+			panic(err)
+		}
+		conn.Close()
 	}()
-	if err := <-receiveStdout; err != nil {
-		return err
-	}
 
-	return nil
-
-}
+	// Stdin
+	_, err = conn.Write([]byte("Hello world"))
+	Expect(err).To(BeNil())
 
-func redirectResponseToOutputStreams(outputStream, errorStream io.WriteCloser, conn io.Reader) (err error) {
 	const (
 		attachPipeStdout = 2
 		attachPipeStderr = 3
 		bufSize          = 8192
 	)
-	reader := bufio.NewReader(conn)
-	buf := make([]byte, 0, bufSize)
 
-	for i := 0; i < 2; i++ {
-		n, err := io.ReadFull(reader, buf[:cap(buf)])
-		buf = buf[:n]
-		if err != nil {
-			if err == io.EOF {
-				break
-			}
-			if err != io.ErrUnexpectedEOF {
-				fmt.Fprintln(os.Stderr, err)
-				break
-			}
-		}
+	return nil
 
-		fmt.Printf("Read %v bytes\n", n)
-		if n > 0 {
-			var dst io.Writer
+	/*
```

Comment:
eventually when we figure out this code, I think it should live in client. Ideally, we could provide an attach that the caller gives channels (And some other args, probably in a struct) and the client wires everything up

Reply:
Yeah we can follow-up on that as well :+1: